synapse/daemon.py

```diff
@@ -478,6 +495,17 @@ def _getTaskFiniMesg(self, task, valu):
                 ('task:fini', {'task': task, 'retn': retn})
             )
 
+    def _reapTasks(self):
+        if self.isfini:
+            return
+        try:
+            self._shareLoopTasks = [task for task in self._shareLoopTasks if not task.done()]
```

Comment:
It looks like you're missing a step here. Don't you want to call `result()` on the done tasks?

Reply:
Also, instead of explicitly reaping, would be easier to use https://docs.python.org/3/library/asyncio-task.html#asyncio.Future.add_done_callback on each future as it is being made. Then we don't need any looping.